server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -958,6 +1025,28 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
             if (format != null) {
                 throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] doesn't support formats.");
             }
+            if (this.script != null) {
+                // values don't live in source - either pull from dv, or re-calculate
+                if (hasDocValues()) {
+                    return new DocValueFetcher(DocValueFormat.RAW, context.getForField(this));
+                }
+                return new ValueFetcher() {
```

Comment:
do we have tests for this scenario?

Reply:
There are specific YAML tests for dv and no-dv cases. Unit testing is trickier but I may be able to make use of some of the bits that Nik plans to add in #70278